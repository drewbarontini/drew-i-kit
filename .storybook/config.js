import { configure, addDecorator } from '@storybook/react';
import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import 'normalize.css';

import { theme } from '../config';

import GlobalStyles from '../components/GlobalStyles';

const req = require.context('../components/', true, /.story.js$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

function withStyledComponents(story) {
  return (
    <ThemeProvider theme={theme}>
      <Fragment>
        {story()}
        <GlobalStyles />
      </Fragment>
    </ThemeProvider>
  );
}

addDecorator(withStyledComponents);
configure(loadStories, module);
