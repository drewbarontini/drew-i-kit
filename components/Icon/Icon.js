import React from 'react';
import PropTypes from 'prop-types';

import icons from '../../config/icons';

import Container from './styled/Container';

const Icon = ({
  name,
  color,
  height = '20',
  size = '20',
  width = '20',
  ...props
}: Props = {}) => {
  const newWidth = size || width;
  const newHeight = size || height;

  return (
    <Container
      color={color}
      width={newWidth}
      height={newHeight}
      viewBox="0 0 20 20"
      {...props}
    >
      {Array.isArray(icons[name]) ? (
        icons[name].map((d, i) => <path key={i} d={d} />)
      ) : (
        <path d={icons[name]} />
      )}
    </Container>
  );
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  height: PropTypes.string,
  size: PropTypes.string,
  width: PropTypes.string,
};

Icon.displayName = 'Icon';

export default Icon;
