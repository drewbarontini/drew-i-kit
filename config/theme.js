const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
