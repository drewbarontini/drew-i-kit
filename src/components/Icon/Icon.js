import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';

import StyledIcon from './styled/Icon';

const Icon = ({ color, height, name, size, theme, width, ...props }) => {
  const newWidth = size || width || theme.defaults.iconSize;
  const newHeight = size || height || theme.defaults.iconSize;

  return (
    <StyledIcon
      color={color}
      width={newWidth}
      height={newHeight}
      viewBox={`0 0 ${theme.defaults.iconSize} ${theme.defaults.iconSize}`}
      {...props}
    >
      {Array.isArray(theme.icons[name]) ? (
        theme.icons[name].map((d, i) => <path key={i} d={d} />)
      ) : (
        <path d={theme.icons[name]} />
      )}
    </StyledIcon>
  );
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  height: PropTypes.string,
  size: PropTypes.string,
  width: PropTypes.string,
};

Icon.defaultProps = {
  color: null,
  height: null,
  size: null,
  width: null,
};

Icon.displayName = 'Icon';

export default withTheme(Icon);
