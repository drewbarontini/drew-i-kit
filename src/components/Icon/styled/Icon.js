import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp, responsive } from '../../../lib';

const Icon = styled.svg`
  align-self: center;
  display: inline-flex;
  vertical-align: middle;

  ${setProp({
    prop: 'color',
    cssProp: 'fill',
    themeKey: 'colors',
    fallback: 'currentColor',
  })};
  ${setProp({
    prop: 'marginLeft',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'marginRight',
    themeKey: 'spacing',
  })};

  ${responsive};
`;

Icon.propTypes = {
  color: PropTypes.string,
  marginLeft: PropTypes.string,
  marginRight: PropTypes.string,
};

Icon.displayName = 'StyledIcon';

export default Icon;
