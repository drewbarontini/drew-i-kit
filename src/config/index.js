export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
