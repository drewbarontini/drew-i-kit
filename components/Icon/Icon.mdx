---
name: Icon
route: /components/icon
---

# Icon

import { Playground } from 'docz';
import icons from '../../config/icons';
import Icon from './';

<Playground>
  <div>
    {icons.map((icon, index) => (
      <Icon name={icon.name} marginRight="s" />
    ))}
  </div>
</Playground>
