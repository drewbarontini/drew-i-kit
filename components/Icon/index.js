import withThemeProvider from '../withThemeProvider';
import Icon from './Icon';

export default withThemeProvider(Icon);
