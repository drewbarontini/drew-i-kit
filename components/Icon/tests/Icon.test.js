import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Icon from '../';

describe('<Icon />', () => {
  it('should render properly', () => {
    const { container, debug } = render(<Icon name="circle" theme={theme} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('fill', 'currentColor');
  });
});
