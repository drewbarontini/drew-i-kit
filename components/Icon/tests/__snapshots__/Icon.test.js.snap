// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render properly 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  fill: currentColor;
}

<svg
  class="c0"
  height="20"
  viewBox="0 0 20 20"
  width="20"
>
  <path
    d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0L10,0z M10,18 c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S14.4,18,10,18L10,18z"
  />
</svg>
`;
